tests/addresses_test.js

```diff
@@ -22,6 +22,107 @@ test("list addresses", async function (t) {
   t.assert(response.status === 200);
 });
 
+test("list addresses' parameters", async function (t) {
+  const list = async (body) => {
+    const response = await prism.setup().then((client) =>
+      client.get(`${resource_endpoint}?${body}`, {
+        headers: prism.authHeader,
+      })
+    );
+    t.assert(response.status === 200);
+    return response.data;
+  };
+
+  /* ################## LIMIT ################## */
+
+  const limit_body = new URLSearchParams({ limit: 8 });
+
+  const limit_response = list(limit_body.toString());
+
+  /* ################## BEFORE ################## */
+
+  const before_body = new URLSearchParams({
+    before:
+      "eyJkYXRlT2Zmc2V0IjoiMjAyMS0wNy0yMFQxOTozMDoyOC4wNTJaIiwiaWRPZmZzZXQiOiJhZHJfNzQyN2FhZWE0MjVhZTlmMCJ9",
+  });
+
+  const before_response = list(before_body.toString());
+
+  /* ################## AFTER ################## */
+
+  const after_body = new URLSearchParams({
+    after:
+      "eyJkYXRlT2Zmc2V0IjoiMjAyMS0wNy0yMFQxOTozMTowOS4wMjRaIiwiaWRPZmZzZXQiOiJhZHJfNmE5MGY5MzNiODkyNDk4OCJ9",
+  });
+
+  const after_response = list(after_body.toString());
+
+  /* ################## INCLUDE ################## */
+
+  const include_body = new URLSearchParams({ "include[]": "total_count" });
+  const include_response = list(include_body.toString());
+
+  /* ################## METADATA ################## */
+
+  const metadata_body = new URLSearchParams({ "metadata[name]": "Harry" });
+  const metadata_response = list(metadata_body.toString());
+
+  /* ################## DATE_CREATED ################## */
+
+  let deepObj = {
+    date_created: { gte: "2021-07-19", lte: "2021-07-20T19:32:00.000Z" },
+  };
+
+  const date_body = new URLSearchParams({});
+  for (const key in deepObj) {
```

Comment:
I might be missing something here, but why create an outer map when the inner map only has one key? You could just hardcode this without using nested loops (it might shave off a few milliseconds from the test run, but it's not super necessary, so it's your call).

Reply:
Ahaha I copy pasted this from the URLencoded tests, where stuff like `metadata` could have multiple keys. Shaving now...